daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -181,3 +190,41 @@ getDispatchEnv Env{..} = do
            , (sdkVersionEnvVar, maybe "" versionToString envSdkVersion)
            , (damlAssistantEnvVar, unwrapDamlAssistantPath envDamlAssistantPath)
            ] ++ filter ((`notElem` damlEnvVars) . fst) originalEnv
+
+
+-- | Auto-installs requested version if it is missing and updates daml-assistant
+-- if it is the latest stable version.
+autoInstall :: DamlPath -> Maybe SdkVersion -> IO (Maybe SdkPath)
+autoInstall damlPath sdkVersionM = do
+    damlConfigE <- try $ readDamlConfig damlPath
+    let doAutoInstallME = queryDamlConfig ["auto-install"] =<< damlConfigE
+        doAutoInstallM  = either (const (Just True)) id doAutoInstallME
```

Comment:
`either (const True) (fromMaybe True) doAutoInstallME` seems simpler than doing it in two steps.

Reply:
managed to make it even simpler :-)